app/routers/calendar_grid.py

```diff
@@ -0,0 +1,136 @@
+
+
+class DayWeekend(Day):
+    def __init__(self, date: datetime):
+        super().__init__(date)
+        self.css = {
+            'div': 'day ',
+            'date': 'day-number t-gray',
+            'daily_event': 'month-event',
+            'daily_event_front': 'daily front bg-warmyellow',
+            'daily_event_back': 'daily back t-darkblue bg-lightgray',
+            'event': 'event',
+        }
+
+
+class Today(Day):
+    def __init__(self, date: datetime):
+        super().__init__(date)
+        self.css = {
+            'div': 'day t-darkblue bg-yellow',
+            'date': 'day-number',
+            'daily_event': 'month-event',
+            'daily_event_front': 'daily front t-lightgray bg-darkblue',
+            'daily_event_back': 'daily back t-darkblue bg-lightgray',
+            'event': 'event',
+        }
+
+
+class FirstDayMonth(Day):
+    def __init__(self, date: datetime):
+        super().__init__(date)
+        self.css = {
+            'div': 'day t-darkblue bg-lightgray',
+            'date': 'day-number',
+            'daily_event': 'month-event',
+            'daily_event_front': 'daily front t-lightgray bg-red',
+            'daily_event_back': 'daily back t-darkblue bg-lightgray',
+            'event': 'event',
+        }
+
+    def __str__(self) -> str:
+        return self.date.strftime("%d %b %y").upper()
+
+
+def create_day(day: datetime) -> Day:
+    """Return the currect day object according to given date."""
+    if Day.is_weekend(day):
+        return DayWeekend(day)
+    if day == date.today():
+        return Today(day)
+    if int(day.day) == 1:
+        return FirstDayMonth(day)
+    return Day(day)
+
+
+def get_next_date(date: datetime) -> Generator[Day, None, None]:
+    """Generate date objects from a starting given date."""
+    yield from (
+        create_day(date + timedelta(days=i))
+        for i in itertools.count(start=1)
+    )
+
+
+def get_date_before_n_days(date: datetime, n: int) -> datetime:
+    """Returns the date before n days."""
+    return date - timedelta(days=n)
+
+
+def get_first_day_month_block(date: datetime) -> datetime:
+    """Returns the first date in a month block of given date."""
+    return list(CALENDAR.itermonthdates(date.year, date.month))[0]
+
+
+def get_n_days(date: datetime, n: int) -> Iterator[Day]:
+    """Generate n dates from a starting given date."""
+    next_date_gen = get_next_date(date)
+    yield from itertools.islice(next_date_gen, n)
+
+
+def split_list_to_lists(dates: List[Any], length: int) -> List[List[Any]]:
+    """Return a 2D list with inner lists length of given size."""
+    return [dates[i:i + length] for i in range(0, len(dates), length)]
+
+
+def get_month_block(date: datetime, n: int = DISPLAY_BLOCK) -> List[List[Day]]:
+    """Returns a 2D list represent a n days calendar from current month."""
+    start = create_day(get_first_day_month_block(date))
+    cal = list(get_n_days(start.date, n))
```

Comment:
Prefer:
`cal = [start] + list(...`

`cal.insert(0` is O(n), where n is the number of elements in `cal`